src/main/java/edu/kit/kastel/sdq/artemis4j/grading/autograder/AutograderRunner.java

```diff
@@ -0,0 +1,61 @@
+/* Licensed under EPL-2.0 2024. */
+package edu.kit.kastel.sdq.artemis4j.grading.autograder;
+
+import de.firemage.autograder.core.CheckConfiguration;
+import de.firemage.autograder.core.Linter;
+import de.firemage.autograder.core.LinterException;
+import de.firemage.autograder.core.LinterStatus;
+import de.firemage.autograder.core.Problem;
+import de.firemage.autograder.core.compiler.JavaVersion;
+import de.firemage.autograder.core.file.TempLocation;
+import de.firemage.autograder.core.file.UploadedFile;
+import edu.kit.kastel.sdq.artemis4j.grading.Assessment;
+import edu.kit.kastel.sdq.artemis4j.grading.ClonedProgrammingSubmission;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public final class AutograderRunner {
+	private AutograderRunner() {
+	}
+
+	public static AutograderStats runAutograder(Assessment assessment, ClonedProgrammingSubmission submission, Locale locale, int threads,
+			Consumer<String> statusConsumer) throws AutograderFailedException {
+		if (!assessment.getSubmission().equals(submission.getSubmission())) {
+			throw new IllegalArgumentException("The assessment and submission do not match");
+		}
+
+		var problemTypesMap = assessment.getConfig().getMistakeTypes().stream().filter(m -> m.getAutograderProblemTypes() != null)
+				.flatMap(m -> m.getAutograderProblemTypes().stream().map(p -> Map.entry(p, m))).distinct()
+				.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+		var checkConfiguration = CheckConfiguration.fromProblemTypes(new ArrayList<>(problemTypesMap.keySet()));
+
+		try (TempLocation tempLocation = TempLocation.of(".autograder-tmp")) {
+			Linter autograder = Linter.builder(locale).threads(threads).tempLocation(tempLocation).maxProblemsPerCheck(100).build();
```

Comment:
Why is the `maxProblemsPerCheck` set to `100`? One could use `-1` so it does not minimize them.

Reply:
It's a placeholder; forgot to change the value